test/test_swallow.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (c) 2021 Jeroen Wijenbergh
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from libqtile import config
+from libqtile.backend.x11 import xcbq
+from libqtile.backend.x11.core import Core
+from libqtile.confreader import Config
+from libqtile.lazy import lazy
+
+
+# Function that does nothing
+def swallow_nop(qtile):
+    pass
```

Comment:
Idea: you could use a global variable as a counter and increment it every time this function is called. That way we can confirm that the bound function is being called when expected.

Reply:
Good idea!

Reply:
I couldn't get a global variable working :/ I don't think pytest likes that, I tried a bunch of things. But the global variable is not correctly updated, it stays at 0. Even though the function is executed

Reply:
Can you make a fixture and extend the scope of the fixture to `module` - your noop function could then increment a value within that fixture.

Feel free to ignore this as I'm coming at it with virtually no context!

Reply:
Oh yeah nice, good idea

Reply:
Thanks. I tried something with fixtures but I hardly understood them. I will have another look at it. I guess the same could be done for the .when() Match pr, but there I just used `toggle_floating` with a test window.

Reply:
@elParaguayo do you have a simple example how can I implement this with fixtures? I can't seem to access it inside the lazy function. How would the lazy function and fixture itself look like?